src/main/java/io/antmedia/rest/StreamsSourceRestService.java

```diff
@@ -440,6 +437,66 @@ public Result moveRight(@ApiParam(value = "the id of the IP Camera", required =
 		return new Result(result);
 	}
 
+	@ApiOperation(value = "Start mp4 muxing", notes = "", response = Result.class)
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/mp4muxing/start/{streamId}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Result startMp4Muxing(@ApiParam(value = "id of the stream", required = true) @QueryParam("streamId") String streamId) {
+		boolean result = false;
+		MuxAdaptor muxAdaptor = getMuxAdaptor(streamId);
+		if(muxAdaptor != null){
+			Mp4Muxer mp4Muxer = getMp4Muxer(muxAdaptor);
+			if(mp4Muxer == null){
+				//avoid multiple call of rest api adding new mp4muxers
+				muxAdaptor.startRecording();
+			}
+			result = true;
+		}
+		return new Result(result);
+	}
+
+	@ApiOperation(value = "Stop mp4 muxing", notes = "", response = Result.class)
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/mp4muxing/stop/{streamId}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Result stopMp4Muxing(@ApiParam(value = "id of the stream", required = true) @QueryParam("streamId") String streamId) {
+		boolean result = false;
+		MuxAdaptor muxAdaptor = getMuxAdaptor(streamId);
+		if(muxAdaptor != null){
+			Mp4Muxer mp4Muxer = getMp4Muxer(muxAdaptor);
+			if(mp4Muxer != null){
+				//avoid multiple call of rest api stopping mp4 muxer
+				muxAdaptor.stopRecording();
+			}
+			result = true;
+		}
+		return new Result(result);
+	}
+
+	@Nullable
+	private Mp4Muxer getMp4Muxer(MuxAdaptor muxAdaptor) {
+		Mp4Muxer mp4Muxer = null;
+		for(Muxer muxer:muxAdaptor.getMuxerList()){
+			if(muxer instanceof Mp4Muxer){
+				mp4Muxer = (Mp4Muxer) muxer;
+			}
+		}
+		return mp4Muxer;
+	}
+
+	@Nullable
+	private MuxAdaptor getMuxAdaptor(@ApiParam(value = "id of the stream", required = true) @QueryParam("streamId") String streamId) {
```

Comment:
This method works for stream sources. Additionally we need to enable this feature both RTMP and WebRTC ingesting as well.